src/transformStateWithClones.js

```diff
@@ -1,13 +1,33 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
```

Comment:
It's good that you're creating a new state to avoid mutating the original state object, but this should be done inside the loop to ensure each action works with the correct previous state. As it stands, all actions will mutate this `newState`, which will lead to incorrect results.